File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/picker_dialog_time/TimeSelection.java
package com.github.rooneyandshadows.lightbulb.dialogs.picker_dialog_time;

import com.github.rooneyandshadows.lightbulb.dialogs.base.LightBulbDialogSelection;

import java.util.Arrays;

class TimeSelection extends LightBulbDialogSelection<int[]> {
        public TimeSelection(int[] current, int[] draft) {
            super(current, draft);
        }

        @Override
        public boolean compareValues(int[] v1, int[] v2) {
            return Arrays.equals(v1, v2);
        }

        @Override
        public boolean hasCurrentSelection() {
            return getCurrentSelection() != null;
        }

        @Override
        public boolean hasDraftSelection() {
            return getDraftSelection() != null;
        }

    }

File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/picker_dialog_time/TimeSelection.kt
package com.github.rooneyandshadows.lightbulb.dialogs.picker_dialog_time

import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogSelection
import java.util.*

internal class TimeSelection(current: IntArray?, draft: IntArray?) : BaseDialogSelection<IntArray?>(current, draft) {
    override fun compareValues(v1: IntArray?, v2: IntArray?): Boolean {
        return Arrays.equals(v1, v2)
    }

    override fun hasCurrentSelection(): Boolean {
        return currentSelection != null
    }

    override fun hasDraftSelection(): Boolean {
        return draftSelection != null
    }
}